dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/changelog/ChangeLogServiceTest.java

```diff
@@ -0,0 +1,311 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.export.changelog;
+
+import static org.hisp.dhis.tracker.Assertions.assertNoErrors;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.IOException;
+import java.util.List;
+import org.hisp.dhis.common.CodeGenerator;
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.common.UID;
+import org.hisp.dhis.dataelement.DataElement;
+import org.hisp.dhis.feedback.NotFoundException;
+import org.hisp.dhis.program.Event;
+import org.hisp.dhis.tracker.TrackerTest;
+import org.hisp.dhis.tracker.imports.TrackerImportParams;
+import org.hisp.dhis.tracker.imports.TrackerImportService;
+import org.hisp.dhis.tracker.imports.domain.TrackerObjects;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.user.UserService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class ChangeLogServiceTest extends TrackerTest {
+
+  @Autowired private ChangeLogService changeLogService;
+
+  @Autowired private TrackerImportService trackerImportService;
+
+  @Autowired private IdentifiableObjectManager manager;
+
+  @Autowired protected UserService _userService;
+
+  private User importUser;
+
+  private TrackerImportParams importParams;
+
+  @Override
+  protected void initTest() throws IOException {
+    userService = _userService;
+    setUpMetadata("tracker/simple_metadata.json");
+    importUser = userService.getUser("M5zQapPyTZI");
+    importParams = TrackerImportParams.builder().userId(importUser.getUid()).build();
+    assertNoErrors(
+        trackerImportService.importTracker(
+            importParams, fromJson("tracker/event_and_enrollment.json")));
+  }
+
+  @Test
+  void shouldFailWhenEventDoesNotExist() {
+    assertThrows(
+        NotFoundException.class,
+        () -> changeLogService.getEventChangeLog(UID.of(CodeGenerator.generateUid())));
+  }
+
+  @Test
+  void shouldFailWhenEventOrgUnitIsNotAccessible() {
+    injectSecurityContextUser(manager.get(User.class, "o1HMTIzBGo7"));
```

Comment:
A helper like

```suggestion
    testAsUser("o1HMTIzBGo7");
```

might be nice